src/main/java/uk/ac/ebi/eva/contigalias/service/AssemblyService.java

```diff
@@ -58,66 +60,56 @@ public AssemblyService(AssemblyRepository repository, @Qualifier("NCBIDataSource
 
     public List<AssemblyEntity> getAssemblyOrFetchByAccession(String accession) throws IOException {
 
-        List<AssemblyEntity> entities = getAssemblyByAccession(accession);
+        Page<AssemblyEntity> entities = getAssemblyByAccession(accession, DEFAULT_PAGE_REQUEST);
         if (!entities.isEmpty()) {
-            return entities;
+            return entities.get().collect(Collectors.toList());
         }
         fetchAndInsertAssembly(accession);
 
-        entities = getAssemblyByAccession(accession);
+        entities = getAssemblyByAccession(accession, DEFAULT_PAGE_REQUEST);
         if (!entities.isEmpty()) {
-            return entities;
-        } else return new LinkedList<>();
+            return entities.get().collect(Collectors.toList());
+        } else {
+            return new LinkedList<>();
+        }
     }
 
-    public List<AssemblyEntity> getAssemblyByGenbank(String genbank) {
-        Optional<AssemblyEntity> entity = repository.findAssemblyEntityByGenbank(genbank);
-        return convertOptionalToList(entity);
+    public Page<AssemblyEntity> getAssemblyByGenbank(String genbank, Pageable request) {
```

Comment:
here I understand your concerns about testing but I would rather have a List here. The reason for having a list instead of an Optional is to make tests easier because the controller was doing too many things, right?

Reply:
Created a new issue for this #52